E-46 feat(ui): add Material UI and initial theme setup

MsSanli opened the pull request:
- Installed Material UI
- Added Poppins font (see already existing index.html file)
- Modifications in index.css and commenting out any MUI conflicts like color scheme/buttons
- Replaced colors with our decided theme

MsSanli commented:
note: Should we consider approving @fsanchez1397's pull request first to avoid merge conflicts or will that not be a problem?

jasminepvo commented:
@MsSanli thanks for submitting the PR! Quick reminder that we’re following the commit structure outlined in the README for consistency. Would you mind updating your PR title to the format and future commits to match the format? Let me know if you need any help!

For reference, the structure is:
<type>(optional scope): <description>

Example:
feat(ui): add Material UI and initial theme setup

jasminepvo commented on package.json:
```diff
@@ -11,6 +11,10 @@
     "test": "vitest"
   },
   "dependencies": {
+    "@emotion/react": "^11.14.0",
+    "@emotion/styled": "^11.14.0",
```
question: this looks good! Just confirming—are we planning to use Emotion for styling instead of regular CSS files, or will we mix both approaches? Lets make sure the team is kept in the loop - either update the frontend channel or note this in the PR

MsSanli replied:
After doing some research, it looks like the latest version of Material UI automatically comes with Emotion. 
Pros: Ability to add inline styles directly to components (sx prop). This is new to a lot of us, but I'm not opposed to it. CSS can coexist as well so we can keep CSS files if needed. 

@fsanchez1397 shared screen w/ @eddiemaceda11 and I to see what would happen with MUI installed, and testing on a button. Commenting out these 2 lines made no difference, but open to the group's thoughts and will update.

jasminepvo replied:
sx prop is typically best for dynamic styling, while css is better for reusability. Let's stick with css since everyone is already familiar with it :)

jasminepvo replied:
To remove: `npm uninstall @emotion/react @emotion/styled`

MsSanli replied:
I see. Thank you! Removed :)

jasminepvo commented on src/index.css:
```diff
@@ -1,11 +1,11 @@
 
-  color-scheme: light dark;
+  /* color-scheme: light dark;
   color: rgba(255, 255, 255, 0.87);
-  background-color: #242424;
+  background-color: #242424; */
```
issue: there are some commented-out styles in index.css that we should clean up, remove the appropriate styling that is being replaced with the theme

jasminepvo commented on src/index.css:
```diff
@@ -15,27 +15,28 @@
   margin: 0;
-  display: flex;
-  place-items: center;
+  /* display: flex; */
+  /* place-items: center; */
```
issue: there are some commented-out styles in index.css that we should clean up, remove the appropriate styling that is being replaced with the theme

jasminepvo commented on src/index.css:
```diff
@@ -15,27 +15,28 @@
+  font-family: 'Poppins', sans-serif;
 }
 
-button {
```
issue: there are some commented-out styles in index.css that we should clean up, remove the appropriate styling that is being replaced with the theme

jasminepvo commented on src/theme/theme.ts:
praise: The MUI setup looks great, and the color palette is well-implemented!

jasminepvo commented on src/app/App.css:
```diff
@@ -12,10 +12,10 @@
 }
 .logo:hover {
-  filter: drop-shadow(0 0 2em #646cffaa);
+  filter: drop-shadow(0 0 2em #1a3b5daa);
 }
 .logo.react:hover {
-  filter: drop-shadow(0 0 2em #61dafbaa);
+  filter: drop-shadow(0 0 2em #28a2a2aa);
 }
```
question: I noticed the color #1a3b5daa / #28a2a2aa has an extra alpha value—just confirming, is the transparency intentional for the drop-shadow effect? Doesn't look like its being used in our project. What do you think? We should clean this up to avoid unnecessary styles along with any other boilerplate from the vite react template

MsSanli replied:
I think if we want to keep things uniform, we can keep the current styles we already have. Thank you

jasminepvo commented:
> note: Should we consider approving @fsanchez1397's pull request first to avoid merge conflicts or will that not be a problem?

@MsSanli Shouldn't be conflict since Felipe was working on a new component - but might be conflict with Eddie's PR that was merged in pertaining to project structure. Let me know if you have trouble rebasing and we can walk through it together

MsSanli commented:
Thank you for the feedback! I've made the following changes:

1. Cleaned up index.css by completely removing commented-out lines
2. Removed the alpha transparency values from the logo hover effects in App.css
3. Removed other unused boilerplate CSS from Vite template

jasminepvo reviewed (approved):
praise: thanks for making all the changes and the research you did! LGTM just make sure to rebase with main before merging :)

jasminepvo commented:
Tagged frontend team so everyone has visibility and understanding of the MUI theme changes

eddiemaceda11 reviewed (approved):
@MsSanli LGTM, nice job!
